src/vector/feature.rs

```diff
@@ -177,6 +177,99 @@ impl<'a> Feature<'a> {
         }
     }
 
+    /// Get the index of the named field.
+    ///
+    /// If the field is missing, returns [`GdalError::InvalidFieldName`].
+    ///
+    fn field_idx_from_name(&self, field_name: &str) -> Result<i32> {
+        let c_str_field_name = CString::new(field_name)?;
+        let field_id =
+            unsafe { gdal_sys::OGR_F_GetFieldIndex(self.c_feature, c_str_field_name.as_ptr()) };
+        if field_id == -1 {
+            return Err(GdalError::InvalidFieldName {
+                field_name: field_name.to_string(),
+                method_name: "OGR_F_GetFieldIndex",
+            });
+        }
+
+        Ok(field_id)
+    }
+
+    /// Get the value of the specified field as a [`i32`].
+    ///
+    /// If the field is missing, returns [`GdalError::InvalidFieldName`].
+    ///
+    /// Returns `Ok(None)` if the field is null.
+    /// Returns `Ok(0)` on other kinds of errors.
+    ///
+    pub fn get_field_integer(&self, field_name: &str) -> Result<Option<i32>> {
```

Comment:
i think a better name would be `field_as_integer`

Reply:
3f7cee4